### ISSUE TITLE
Client Groups and Settings

### ISSUE THREAD

**Issue by sullimander** [AdguardTeam/AdGuardHome#2527]
### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Problem Description

I would like to group clients together and set rules for the client group and not have to do it for each client. For example, I want to set different rule for my devices, for my wife's devices, for my kids devices, and had the default rules apply to all other devices. I know I can do this by setting it on each client, but that is a lot to manage, when I want to set the same rules for all clients in a group.

### Proposed Solution

Allow for creating custom groups of clients, and apply custom settings to the group as a whole. Ideally this would be very similar to the settings on the Client Config screen.

**Comment by ameshkov** [comment]
Note that one client can have multiple identifiers, this would partly solve the issue.

**Comment by Aikatsui** [comment]
@sullimander this is a trouble with word but it is more suitable. Client same as client group or profile, you can find more feature requests that awaiting for implement to make this component more flexible

**Comment by sullimander** [comment]
I like that I can "group" clients by assigning multiple MAC addresses to the same client, I've done this with my laptop assigning wifi and ethernet to the same client. But it would be really nice to create a kids group that has all my kids devices in it. That way it's easy to see the data for each device, but I can have a filter for them configured at the group level. It's a little hard if you have multiple devices registered as the same client to figure out which is which. They all show as "Client Name (IP Address)" in the reporting, so I have to goto the DCHP reporting to know which device is which.

I also have a use case where my son plays a mobile game that has ads on it that make a big difference to his game play (gives free lives, etc.). I don't want to unblock it on all his devices, just the iPad he uses. I know there are ways to do that with the existing options, but the easiest would be to allow me to specify those rules just for the one device using the configured client name for the device but the other group rules would still apply.

Hopefully that all makes sense. I've been really happy with AdGuard Home, I've been using it as my DNS and DHCP server for over a month now, and it's been working great. Keep up the amazing work!

**Comment by fermulator** [comment]
This is a big gap for me in transitioning away from pihole

https://docs.pi-hole.net/group_management/example/

**Comment by fermulator** [comment]
client tags maybe come close ... https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#ctag

but we can't have custom tags?

**Comment by mister-awesome** [comment]
I fully agree with the other comments here; this tool is nearly perfect; please consider adding this group management functionality.
Or otherwise, please let us know why you didn't address developing this feature yet?
Thanks!

**Comment by emanzx** [comment]
+1 for this features...

**Comment by ados8** [comment]
Really miss the feature coming from Pihole and the only thing drawing me back.
About the only missing feature missing.

**Comment by barbuslex** [comment]
+1 need groups for different devices rules

**Comment by chrgeorgeson** [comment]
Fully agree. We need a way to manage groups of clients. I absolutely love Adguard Home but can't use it because I need to be able to manage my kid's devices via groups. Otherwise, I spend too much time in the UI going to individual clients. 

We have multiple computers, phones, iPads, and TVs to manage.

**Comment by sebibasti0815** [comment]
I would also like to have support for user groups, especially for blocked services. I don't currently need this for filter lists.

That's roughly how I imagine it:
All: *
People: Dad, Mom, Kids
Kids: F, G
IT-Support: Dad
Homeoffice: Mom

I could then assign the hardest rules for "All" (lots of blocked services) and use them to prevent IoT devices from doing nonsense, for example. I could then unblock most things for "People", and I can set special rules for the "Kids" (perhaps time-controlled in the future, something like this: Facebook, Instagram and X from 5 to 7 pm) and for the homeoffice working Mom who needs Instagram...

**Comment by jeitnier** [comment]
4.5 years later and nothing...seems like this isn't ever going to happen...

**Comment by chrgeorgeson** [comment]
> 4.5 years later and nothing...seems like this isn't ever going to happen...

Yup! It's crazy to me that management isn't included for devices.

**Comment by cringerjs** [comment]
OMG it's 2025 devs.  This request is a must in today's technical landscape.  In most households each user has multiple devices and having to configure each client is really a big joke.  Think about it, even Comcast/xfinity has this specific feature built into their gateways.  Even a whole parental control that can filter some of this stuff.   The fact that we can't even get it on the roadmap for this application is just absurd.

**Comment by lgaudreau** [comment]
> OMG it's 2025 devs. 

No response more than 3 months later...

**Comment by jeitnier** [comment]
Made the switch back to pihole pretty much for this reason

**Comment by chrgeorgeson** [comment]
Same. I like pihole but wish I could use Adguard.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by VsevolodNaumov** [comment]
Thanks for the request. I’ve shared it with the development team as a feature request for consideration. We’ll update this issue if there’s any progress.

**Comment by Generalnoob** [comment]
It's strange that such a simple feature request can still be on the wish list in 2026 when this ticket was started in 2021... why has this not been implemented yet?

**Comment by Sil3ntVip3r** [comment]
I traced this against current master and the active related work.

[#8274](https://github.com/AdguardTeam/AdGuardHome/pull/8274) with its UI companion [#8275](https://github.com/AdguardTeam/AdGuardHome/pull/8275) is a useful incremental step: it lets an individual client select its own filter and allow lists. It does **not** introduce reusable client groups, inherited group settings, or arbitrary policy tags, so it should not be treated as resolving this issue.

A standalone group implementation needs an agreed MVP contract before a PR can be made safely. Today, a request resolves to one persistent client, and client tags are predefined `$ctag` rule selectors rather than user-defined shared policies. The broader custom-filtering work also has engine-memory, atomic-update, and DNS-consistency constraints documented in [#435](https://github.com/AdguardTeam/AdGuardHome/issues/435#issuecomment-1926935842).

Could the first supported scope be clarified on these points?

1. Group scope: only filter-list/`$ctag` policy, or full Client Config inheritance (blocked services, booleans, upstreams, etc.)?
2. Membership: one group or multiple groups per client; is nesting needed?
3. Precedence: global -> ordered groups -> device override?
4. Per-field behavior: inherit, replace, or union for lists, blocked services, and conflicting booleans?
5. Compatibility: expected API/config behavior for group deletion, rename, and legacy clients?

Once that narrow first slice is chosen, it can be implemented and tested in isolated PRs without duplicating #8274/#8275 or locking in an accidental policy model.
